qiskit_experiments/matplotlib.py

```diff
@@ -13,28 +13,28 @@
 Matplotlib helper functions
 """
 
-import functools
+from matplotlib.figure import Figure
+from matplotlib.backends.backend_svg import FigureCanvasSVG
 
-from matplotlib import pyplot
+default_figure_canvas = FigureCanvasSVG  # pylint: disable=invalid-name
+"""Matplotlib canvas to use when rendering a figure. This needs to be a
+canvas for a `non-interactive backend
+<https://matplotlib.org/stable/tutorials/introductory/usage.html#the-builtin-backends>`_.
+The default is `FigureCanvasSVG`."""
 
 
-def requires_matplotlib(func):
-    """Decorator for functions requiring matplotlib"""
+def get_non_gui_ax():
+    """Return a matplotlib axes that can be used in a child thread.
 
-    @functools.wraps(func)
-    def wrapped(*args, **kwargs):
-        # Analysis/plotting is done in a separate thread (so it doesn't block the
-        # main thread), but matplotlib doesn't support GUI mode in a child thread.
-        # The code below switches to a non-GUI backend "Agg" when creating the
-        # plot. An alternative is to run this in a separate process, but then
-        # we'd need to deal with pickling issues.
+    Analysis/plotting is done in a separate thread (so it doesn't block the
+    main thread), but matplotlib doesn't support GUI mode in a child thread.
+    This function creates a separate Figure and attaches a non-GUI
+    SVG canvas to it.
```

Comment:
Maybe document the global variable usage here?

Reply:
This module is not actually included in the doc, so the docstring won't be easy to find. I ended up moving this `matplotlib.py` to `framework` and document the usage in its `__init__`.